token_auth/serializers/register.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+from rest_framework_jwt.serializers import PasswordField
+
+User = get_user_model()
+
+class RegisterSerializer(serializers.ModelSerializer):
+
+    password = PasswordField(
+        write_only=True,
+        required=True
+    )
+    password2 = PasswordField(
+        write_only=True,
+    )
+
+    class Meta:
+        model = User
+        fields = [
+            "username",
+            "email",
+            "password",
+            "password2",
+        ]
+
+    def create(self, validated_data):
+
+        username = validated_data.get('username', None)
+        email = validated_data.get('email', None)
+        password = validated_data.get('password', None)
+        password2 = validated_data.get('password2', None)
+
+        if (email and User.objects.filter(email=email).exclude(username=username).exists()):
+            raise serializers.ValidationError("Email addresses must be unique.")
+
+        if password != password2:
+            raise serializers.ValidationError("The two passwords differ.")
+
+        user = User(username=username, email=email)
+        user.set_password(password)
+        user.save()
+
+        return user
+        
```

Comment:
No newline at the end of the file, but I see a new line. As @Monal5031 said, this could be because of the line endings. Can you check if you're using LF or CRLF?